feat: improved backup system with encryption and scheduling

NathanJ60 opened the pull request:
## Summary

- Add GPG encryption (AES256) for secure backup archives
- Add configurable destination (local directory or remote via rsync/scp)
- Backup all `.env` files recursively (not just backend/frontend)
- Backup `frontend/public/home/` assets
- Add `backup-cron.sh` wrapper for scheduled backups with retention policy
- Add `backup.conf.example` configuration template
- Add optional Docker backup service via compose profile
- Complete documentation rewrite with usage examples

## New Files

| File | Description |
|------|-------------|
| `backup-cron.sh` | Cron wrapper with logging, locking, and retention |
| `backup.conf.example` | Configuration template for scheduled backups |

## Usage Examples

```bash
# Basic backup
./backup.sh --db-docker

# Encrypted backup
./backup.sh --db-docker --encrypt --password "secret"

# Encrypted backup to custom directory
./backup.sh --db-docker --encrypt --password "secret" --destination /backups

# Remote upload
./backup.sh --db-docker --encrypt --password "secret" \
    --remote-type rsync --remote-host backup.example.com \
    --remote-user backup --remote-path /backups/arcadia
```

## Test Plan

- [x] Basic backup creates valid zip archive
- [x] Encrypted backup creates `.tar.gz.gpg` file with AES256
- [x] Decryption works with correct password
- [x] Custom destination moves backup to specified directory
- [x] Cron wrapper runs backup and logs output
- [x] Retention policy deletes old backups

Closes #416

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Full scheduled backup system: database, env files, and frontend assets; optional AES-256 encryption, retention policy, optional remote upload, and Docker or local DB modes; run once or scheduled.

* **Scheduling**
  * Added a cron-compatible scheduler wrapper and a container entrypoint to run backups on a schedule or oneshot.

* **Documentation**
  * Comprehensive backup guide with quick start, encryption workflows, scheduling, retention, remote upload, restore steps, troubleshooting, and examples.

* **Configuration**
  * Added a detailed example configuration template for customizing backup behavior and scheduling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

NathanJ60 committed:
feat: improved backup system with encryption and scheduling

- Add GPG encryption (AES256) for backup archives
- Add configurable destination (local or remote via rsync/scp)
- Backup all .env files recursively
- Backup frontend/public/home/ assets
- Add backup-cron.sh wrapper for scheduled backups
- Add backup.conf.example configuration template
- Add optional Docker backup service (compose profile)
- Update documentation with full usage guide

Closes #416

NathanJ60 committed:
feat: add daemon mode for backup container

- Add backup-entrypoint.sh to handle daemon/oneshot modes
- Add backup-daemon service in compose.yml with internal cron
- Update backup.conf.example with daemon mode documentation
- Container runs crond in foreground for scheduled backups

NathanJ60 committed:
fix: remove :ro from scripts to allow chmod in container

NathanJ60 committed:
docs: add daemon mode documentation for backup

FrenchGithubUser commented:
there should be a single container for the backup, that has an optional deamon mode. Having 2 containers that do almost the same thing is a bit heavy :)

NathanJ60 committed:
refactor: merge backup services into single container

- Combine backup and backup-daemon into one service
- Support both oneshot and daemon modes via BACKUP_MODE env var
- Default to oneshot mode for backward compatibility
- Daemon mode: docker compose --profile backup run -d -e BACKUP_MODE=daemon backup

NathanJ60 committed:
fix: support BACKUP_DESTINATION env var and remove chmod in container

- backup.sh: read BACKUP_DESTINATION from env if set, fallback to '.'
- compose.yml: remove chmod (files mounted read-only, already executable)

NathanJ60 committed:
docs: update backup documentation for single container

- Update commands to use BACKUP_MODE=daemon instead of separate service
- Change container name references from arcadia_backup_daemon to arcadia_backup
- Simplify daemon mode instructions

FrenchGithubUser commented on compose.yml:
```diff
@@ -114,6 +114,38 @@ services:
+    profiles:
+      - backup
+    volumes:
+      - ./backup.sh:/app/backup.sh:ro
```
shouldn't we just mount the whole current folder instead of all those ones separately?

FrenchGithubUser commented on docs/src/backup.md:
```diff
@@ -1,115 +1,309 @@
 # Backup
 
-This page explains how to create backups of your Arcadia installation, including the database and configuration files.
+This page explains how to create backups of your Arcadia installation, including the database, configuration files, and frontend assets.
```
frontend `custom` assets

NathanJ60 committed:
fix: simplify backup container volumes and use /tmp for temp files

- Mount whole project folder instead of individual files
- Use /tmp for backup temp directory (read-only mount compatible)
- Fix entrypoint script path reference
- Update docs wording: frontend assets -> frontend custom assets
